ffmpeg: add v6.1.1 and older patch release updates

aumuell opened the pull request:
adds 6.1.1, 5.1.4, 4.4.4, 3.4.13 and 2.8.22

aumuell committed:
ffmpeg: add v6.1.1

aumuell committed:
ffmpeg: checksum new minor releases

tldahlgren reviewed (dismissed):
Confirmed the 5 new version sha256s.  

@xjrc Are there any other changes needed to support the new versions?

aumuell committed:
ffmpeg: fix build of @6.1.1 with new vulkan-headers

changes between vulkan-headers@1.3.276 and @1.3.279 lead to compile errors,
apply upstream patch

aumuell committed:
ffmpeg: don't apply patches upstreamed in @5.1.4

aumuell committed:
ffmpeg: fix %apple-clang@15 clang-1500.3.9.4

otherwise, versions 2 and 3 error out because of warnings about
incompatible function pointer types

aumuell commented:
Unfortunately, I have to add a few patches:
- to not apply patches to @5.1.4 that are already included in upstream version (apparently I forgot to compile this)
- for new incompatibilities with vulkan headers 1.3.279
- for incompatibilities with Apple clang version 15.0.0 (clang-1500.3.9.4), which comes with Xcode 15.3

aumuell commented:
rebased because of conflicts with binutils patch hash update
